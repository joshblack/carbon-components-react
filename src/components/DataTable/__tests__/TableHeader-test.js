import React from 'react';
import { shallow } from 'enzyme';
import { TableHeader } from '../';

describe('TableHeader', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      children: <div />,
      onClick: jest.fn(),
      onKeyDown: jest.fn(),
      isSortHeader: false,
      sortDirection: null,
    };
  });

  it('should render', () => {
    const wrapper = shallow(<TableHeader {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when it is sorting', () => {
    const wrapper = shallow(
      <TableHeader {...mockProps} isSortHeader={true} sortDirection="ASC" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should support translations', () => {
    expect(TableHeader.translationKeys).toBeDefined();
    const mockTranslateWithId = jest.fn(() => 'hi');
    const wrapper = shallow(
      <TableHeader {...mockProps} translateWithId={mockTranslateWithId} />
    );
    expect(mockTranslateWithId).toHaveBeenCalled();
    expect(mockTranslateWithId).toHaveBeenCalledWith(
      TableHeader.translationKeys[0]
    );
  });
});
