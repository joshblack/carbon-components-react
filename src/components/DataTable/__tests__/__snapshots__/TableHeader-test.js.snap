// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader should render 1`] = `
<th>
  <button
    className="bx--table-sort-v2"
    onClick={[MockFunction]}
    onKeyDown={[MockFunction]}
  >
    <span
      className="bx--table-header-label"
    >
      <div />
    </span>
    <Icon
      className="bx--table-sort-v2__icon"
      description="Sort arrow"
      fillRule="evenodd"
      name="caret--down"
      role="img"
    />
  </button>
</th>
`;

exports[`TableHeader should render when it is sorting 1`] = `
<th>
  <button
    className="bx--table-sort-v2 bx--table-sort-v2--active bx--table-sort-v2--ascending"
    onClick={[MockFunction]}
    onKeyDown={[MockFunction]}
  >
    <span
      className="bx--table-header-label"
    >
      <div />
    </span>
    <Icon
      className="bx--table-sort-v2__icon"
      description="Sort arrow"
      fillRule="evenodd"
      name="caret--down"
      role="img"
    />
  </button>
</th>
`;
